rarible_marketplace_indexer/event/abstract_action.py

```diff
@@ -199,3 +199,360 @@ async def handle(
+                take_contract=dto.take.contract,
+                take_token_id=dto.take.token_id,
+                take_value=dto.take.value,
+            )
+        else:
+            order.last_updated_at = transaction.data.timestamp
+            order.make_value = dto.make.value
+            order.take_value = dto.take.value
+            order.save()
+
+        await ActivityModel.create(
+            type=ActivityTypeEnum.MAKE_BID,
+            network=datasource.network,
+            platform=cls.platform,
+            order_id=order.id,
+            internal_order_id=dto.internal_order_id,
+            maker=dto.maker,
+            make_asset_class=dto.make.asset_class,
+            make_contract=dto.make.contract,
+            make_token_id=dto.make.token_id,
+            make_value=dto.make.value,
+            take_asset_class=dto.take.asset_class,
+            take_contract=dto.take.contract,
+            take_token_id=dto.take.token_id,
+            take_value=dto.take.value,
+            sell_price=dto.make_price,
+            operation_level=transaction.data.level,
+            operation_timestamp=transaction.data.timestamp,
+            operation_hash=transaction.data.hash,
+            operation_counter=transaction.data.counter,
+            operation_nonce=transaction.data.nonce,
+        )
+
+
+class AbstractPutFloorBidEvent(EventInterface):
+    @staticmethod
+    @abstractmethod
+    def _get_floor_bid_dto(
+        transaction: Transaction,
+        datasource: TzktDatasource,
+    ) -> ListDto:
+        raise NotImplementedError
+
+    @classmethod
+    @final
+    async def handle(
+        cls,
+        transaction: Transaction,
+        datasource: TzktDatasource,
+    ):
+        dto = cls._get_floor_bid_dto(transaction, datasource)
+
+        order = await OrderModel.get_or_none(
+            internal_order_id=dto.internal_order_id, network=datasource.network, platform=cls.platform, status=OrderStatusEnum.ACTIVE
+        )
+
+        if order is None:
+            order = await OrderModel.create(
+                network=datasource.network,
+                platform=cls.platform,
+                internal_order_id=dto.internal_order_id,
+                status=OrderStatusEnum.ACTIVE,
+                started_at=transaction.data.timestamp,
+                ended_at=dto.ended_at,
+                make_stock=dto.make.value,
+                salt=transaction.data.counter,
+                created_at=transaction.data.timestamp,
+                last_updated_at=transaction.data.timestamp,
+                make_price=dto.make_price,
+                maker=dto.maker,
+                make_asset_class=dto.make.asset_class,
+                make_contract=dto.make.contract,
+                make_token_id=dto.make.token_id,
+                make_value=dto.make.value,
+                take_asset_class=dto.take.asset_class,
+                take_contract=dto.take.contract,
+                take_token_id=dto.take.token_id,
```

Comment:
Do we have a token_id when we make floor bid?

Reply:
Yes, but since it comes from the DTO it will take the value there (may be None)